drivers/overlay/overlay.go

```diff
@@ -882,6 +882,24 @@ func (d *Driver) UpdateLayerIDMap(id string, toContainer, toHost *idtools.IDMapp
 	if err := idtools.MkdirAs(diffDir, 0755, rootUID, rootGID); err != nil {
 		return err
 	}
+
+	if rootUID != 0 {
+		parent := filepath.Dir(diffDir)
+		for parent != "/" {
+			fi, err := os.Stat(parent)
+			if err != nil {
+				return err
+			}
+
+			const desiredMask = 0711
```

Comment:
I'm still not sure about 711 here.  But if Dan an Nalin think it's OK, then I'm good with it.